With cross compilation there is two points:

1. Cross compilation uses clang-8

compiler version check to include -Wno-ctad-maybe-unsupported is 'OS_DARWIN AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 8 AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 11'

So build should not add this option for clang-8

Can you confirm that cross compilation is working on master?

2. I cannot validate cross compilation is working on master on my setup (Ubuntu 18.0.4). Configuration step is failed.

CMake Error at /usr/local/share/cmake-3.12/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find Threads (missing: Threads_FOUND)
Call Stack (most recent call first):
  /usr/local/share/cmake-3.12/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  /usr/local/share/cmake-3.12/Modules/FindThreads.cmake:205 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  cmake/darwin/default_libs.cmake:29 (find_package)
  CMakeLists.txt:234 (include)

-- Configuring incomplete, errors occurred!

@alexey-milovidov

--- a/contrib/libcxx-cmake/CMakeLists.txt
+++ b/contrib/libcxx-cmake/CMakeLists.txt
@@ -44,7 +44,7 @@ target_include_directories(cxx SYSTEM BEFORE PUBLIC $<BUILD_INTERFACE:${LIBCXX_S
 target_compile_definitions(cxx PRIVATE -D_LIBCPP_BUILDING_LIBRARY -DLIBCXX_BUILDING_LIBCXXABI)
 
 target_compile_options(cxx PUBLIC -nostdinc++ -Wno-reserved-id-macro)
-if (OS_DARWIN AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 8 AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 11)
+if (OS_DARWIN AND (NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9) AND (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 11))
     target_compile_options(cxx PUBLIC -Wno-ctad-maybe-unsupported)
 endif ()
 

